src/Domain/Python/BlockLogicalCoordinates.cpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Python/BlockLogicalCoordinates.hpp"
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/IdPair.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/BlockLogicalCoordinates.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/Structure/BlockId.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace py = pybind11;
+
+namespace domain::py_bindings {
+
+namespace {
+template <size_t Dim>
+void bind_block_logical_coordinates_impl(py::module& m) {  // NOLINT
+  py::class_<
+      IdPair<domain::BlockId, tnsr::I<double, Dim, Frame::BlockLogical>>>(
+      m, ("BlockIdAndLogicalCoord" + get_output(Dim) + "D").c_str());
+  m.def(
+      "block_logical_coordinates",
+      [](const Domain<Dim>& domain,
+         const std::array<DataVector, Dim>& inertial_coords,
```

Comment:
I guess when the tensor bindings are merged this could get updated?

Reply:
done